File: 技巧/169.多数元素.java
class Solution {
    public int majorityElement(int[] nums) {
        int cnt=0,index=0;
        for(int i=0;i<nums.length;i++){ 
            if(cnt==0){
                index=i;
            }
            if(nums[i]!=nums[index]){
                cnt--;
            }else{
                cnt++;
            }
        }
        return nums[index];
    }
}

File: 技巧/169.多数元素.go
func majorityElement(nums []int) int {
	num, cnt := 0, 0
	for _, v := range nums {
		if cnt == 0 {
			num = v
			cnt++
		} else if v == num {
			cnt++
		} else {
			cnt--
		}
	}
	return num
}